gt/main.c

```diff
@@ -88,30 +156,68 @@ gt_proc(void *arg)
 			 *
 			 * Other packets will be fowarded directly.
 			 */
+			rte_pktmbuf_adj(m, sizeof(struct ether_hdr));
 
 			/*
-			 * TODO Lookup the policy decision.
+			 * Lookup the policy decision.
 			 *
 			 * The policy, which is defined by a Lua script,
 			 * decides which capabilities to grant or decline,
 			 * the maximum receiving rate of the granted
 			 * capabilities, and when each decision expires.
+			 *
+			 * Notice that, the packet now only contains L3
+			 * and above information.
 			 */
+			ret = lookup_policy_decision(m,
+				&inner_ip_ver, &policy, instance);
+			if (ret < 0) {
+				rte_pktmbuf_free(m);
+				continue;
+			}
+
+			if (policy.state == GK_GRANTED) {
+				/*
+	 		 	 * Fill up the Ethernet header, and forward
+			 	 * the original packet to the destination.
+	 		 	 */
+				new_eth = (struct ether_hdr *)
+					rte_pktmbuf_prepend(
+					m, sizeof(struct ether_hdr));
+				ether_addr_copy(&gt_conf->net->front.eth_addr,
```

Comment:
Once the packet is granted, grantor server needs to forward the original packet to destination, so the destination MAC address should come from LLS, right? The ``ether_type`` field will be filled up according to the ip version of the packet.